### DISCUSSION TITLE
The MIDI.read() buffer

### DISCUSSION THREAD

**Discussion by RiNoize** [FortySevenEffects/arduino_midi_library#323] [Q&A] [answered]
Hello, can the buffer of MIDI.read() be reduced?
On my Arduino nano it equates to almost 2k of program memory when I turn it on.
I am using MIDI.turnThruOn(); to pass all messages from RX directly to TX.
And I'm using SysEx arrays that are 18 bytes long, no more than that.
Thanks in advance.

**Comment by lathoub** [comment, marked as answer]
The [wiki](https://github.com/FortySevenEffects/arduino_midi_library/wiki/Using-custom-Settings) might help (overriding the midi::DefaultSettings)

**Reply by RiNoize** [reply]
Hello Franky, thank you for your prompt response.
I replaced the statement I use in my sketch:

MIDI_CREATE_INSTANCE(HardwareSerial, Serial, MIDI);

For this other one:

struct MySettings : public midi::DefaultSettings
{
     static const unsigned SysExMaxSize = 32; /
     static const bool UseRunningStatus = true;
};

MIDI_CREATE_CUSTOM_INSTANCE(HardwareSerial, Serial, MIDI, MySettings);

And the Arduino IDE gives me this error:

MIDI_Library\src/serialMIDI.h:62:13: error: 'BaudRate' is not a member of 'midi::SerialMIDI<HardwareSerial, MySettings>::Settings {aka MySettings}' mSerial.begin(Settings::BaudRate);

**Reply by franky47** [reply]
It was @lathoub who gave you the answer 😉

Maybe he can chime in here too, I believe `BaudRate` is only defined in the Serial transport settings, so you'd have to inherit from it instead of the base `DefaultSettings`.

**Reply by lathoub** [reply]
```cpp
struct MySettings : public midi::DefaultSettings
{
  static const unsigned SysExMaxSize = 32;
};

MIDI_CREATE_CUSTOM_INSTANCE(HardwareSerial, Serial, MIDI, MySettings);
```

work fine for the Nano. 
Not sure why the `BaudRate` error is showing up, it is unrelated. Are your libs messed up? (happened to me lately).

@franky47 I have included updated examples and wiki for custom settings in PR #286 for the next release.
